master/internal/resourcemanagers/kubernetes_resource_manager.go

```diff
@@ -214,6 +217,21 @@ func (k *kubernetesResourceManager) receiveRequestMsg(ctx *actor.Context) error
 	case sproto.ResourcesReleased:
 		k.resourcesReleased(ctx, msg.TaskActor)
 
+	case sproto.UpdatePodStatus:
+		var ref *actor.Ref
+		for addr, id := range k.addrToContainerID {
```

Comment:
suggestion: would a reverse map of `addrToContainerID map[*actor.Ref]cproto.ID` work better here?  if it's the only usecase?

Reply:
There will be a trade-off between using a map here and using a map during deletion. I only expect each to be used once per experiment, so the runtime will be the same, so I chose this design where the code is a bit more compact. We could use two maps as an index and reverse index. Thoughts?

Reply:
that make sense I didn't consider deletions

Reply:
I'll actually use a map during Update Pod Status and let deletion be a linear time operation. In the future, we may add finer grained statuses on k8s, in which case the pod status could be updated multiple times per experiment.